rsocket-core/src/main/java/io/rsocket/plugins/InterceptorRegistry.java

```diff
@@ -30,16 +31,54 @@
  * </ul>
  */
 public class InterceptorRegistry {
-  private List<RSocketInterceptor> requesterInteceptors = new ArrayList<>();
-  private List<RSocketInterceptor> responderInterceptors = new ArrayList<>();
+  private List<Supplier<? extends RequestInterceptor>> requesterRequestInterceptors =
+      new ArrayList<>();
+  private List<Supplier<? extends RequestInterceptor>> responderRequestInterceptors =
+      new ArrayList<>();
+  private List<RSocketInterceptor> requesterRSocketInterceptors = new ArrayList<>();
+  private List<RSocketInterceptor> responderRSocketInterceptors = new ArrayList<>();
   private List<SocketAcceptorInterceptor> socketAcceptorInterceptors = new ArrayList<>();
   private List<DuplexConnectionInterceptor> connectionInterceptors = new ArrayList<>();
 
+  /** Add an {@link RequestInterceptor} that will hook into Requester RSocket requests' phases. */
+  public InterceptorRegistry forRequesterRequests(
+      Supplier<? extends RequestInterceptor> interceptor) {
+    requesterRequestInterceptors.add(interceptor);
+    return this;
+  }
+
+  /**
+   * Variant of {@link #forRequesterRequests(Supplier)} with access to the list of existing
+   * registrations.
+   */
+  public InterceptorRegistry forRequesterRequests(
+      Consumer<List<Supplier<? extends RequestInterceptor>>> consumer) {
+    consumer.accept(requesterRequestInterceptors);
+    return this;
+  }
+
+  /** Add an {@link RequestInterceptor} that will hook into Requester RSocket requests' phases. */
+  public InterceptorRegistry forResponderRequests(
+      Supplier<? extends RequestInterceptor> interceptor) {
+    responderRequestInterceptors.add(interceptor);
+    return this;
+  }
+
+  /**
+   * Variant of {@link #forResponderRequests(Supplier)} with access to the list of existing
+   * registrations.
+   */
+  public InterceptorRegistry forResponderRequests(
+      Consumer<List<Supplier<? extends RequestInterceptor>>> consumer) {
+    consumer.accept(responderRequestInterceptors);
+    return this;
+  }
```

Comment:
Since the naming convention is not so much about the type of interceptor but about the what the interceptor is installed on, perhaps we can go with overloaded methods, e.g. `forRequester(Supplier<? extends RequestInterceptor>)`. We could also probably avoid the variants with `Consumer<List<Supplier>>` for now until we know it really is needed?

Reply:
It clashes with the existing one, we can not have Consumer of List of RSocket as well as Consumer of List of RequestInterceptors

Reply:
fixed